File: resiliences-service/interfaces/IUserBankService.cs
using resiliences_service.Models;
using resiliences_service.Payloads;

namespace resiliences_service.interfaces
{
    public interface IUserBankService
    {
        Task CreateBankAsync(UserBankList bank);
        Task<UserBankList?> FindByIdAsync(uint id);
        Task<UserBankList?> FindByAccountNumberAsync(string accountNumber);
        Task<List<UserBankList>> FindByUserIdAsync(uint userId);
        Task DeleteByIdsAsync(List<uint> ids);
        Task<bool> FindByAccountNumberAndBankNameAsync(string accountNumber, string bankName);
        Task<UserBankList?> FindInternalAsync(string accountNumber, string bankCode);
        Task<List<PayStackBankList>> FetchAllBanksAsync();
        Task<PaystackAccountData?> VerifyExternalAsync(string accountNumber, string bankCode);
    }
}

File: common/src/main/java/com/epay/common/interfaces/IUserBankService.java
package com.epay.common.interfaces;

import java.util.List;
import java.util.Optional;
import com.epay.domain.bank.dto.PaystackDtos.PayStackBankList;
import com.epay.domain.bank.dto.PaystackDtos.PaystackAccountData;
import com.epay.domain.bank.entity.UserBankList;
 
public interface IUserBankService {
 
    UserBankList createBank(UserBankList bank);
 
    Optional<UserBankList> findById(Long id);
 
    Optional<UserBankList> findByAccountNumber(String accountNumber);
 
    List<UserBankList> findByUserId(Long userId);
 
    void deleteByIds(List<Long> ids);
 
    boolean findByAccountNumberAndBankName(String accountNumber, String bankName);
 
    Optional<UserBankList> findInternal(String accountNumber, String bankCode);
 
    List<PayStackBankList> fetchAllBanks();
 
    PaystackAccountData verifyExternal(String accountNumber, String bankCode);
}
 